app/models/health/sdh_case_management_note.rb

```diff
@@ -119,6 +119,12 @@ def remove_housing_placement_date
       self.housing_placement_date = nil unless housing_status_includes_date?
     end
 
+    after_save :record_housing_status
```

Comment:
These "after saves" make me a little nervous given that they potentially generate more records which have the potential to touch the associated records.  Do you think it makes sense to add some really basic tests to confirm if you create and then update one of these (or the others that behave similarly) that it doesn't get stuck in some sort of loop?  Might also want to confirm you get the housing status you expected.